src/API/Endpoints/Reports/PaymentMethods.php

```diff
@@ -14,32 +14,59 @@ public function __construct() {
 		$this->endpoint = 'payment-methods';
 	}
 
-	public function get_report($request) {
-
-		$labels = [];
-		$data = [];
+	public function get_report( $request ) {
 
 		// Use give_count_payments logic to get payments
 		$gateways = give_get_payment_gateways();
-		$stats = new \Give_Payment_Stats();
+		$stats    = new \Give_Payment_Stats();
+
+		$gatewaysArr = [];
 
 		foreach ( $gateways as $gateway_id => $gateway ) {
-			$donations = $stats->get_earnings( 0, date($request['start']), date($request['end']), $gateway_id );
-			array_push($labels, $gateway['admin_label']);
-			array_push($data, $donations);
+			$gatewaysArr[] = [
+				'admin_label' => $gateway['admin_label'],
+				'count'       => $stats->get_sales( 0, date( $request['start'] ), date( $request['end'] ), $gateway_id ),
+				'amount'      => $stats->get_earnings( 0, date( $request['start'] ), date( $request['end'] ), $gateway_id ),
+			];
 		}
+		$sorted = usort( $gatewaysArr, [ $this, 'compare_gateways' ] );
```

Comment:
Just use a closure here since it's really a function completely unique to this line.

Reply:
Done, ref: https://github.com/impress-org/givewp/commit/e90d1e69973dc4240946f3ed115fc39600439283